cds_ils/migrator/acquisition/orders.py

```diff
@@ -0,0 +1,225 @@
+    )
+
+
+def get_patron_pid(record):
+    """Get patron_pid from existing users."""
+    anonym = current_app.config["ILS_PATRON_ANONYMOUS_CLASS"](-2)
+    patron_pid = anonym.id
+    user = get_user_by_legacy_id(record["id_crcBORROWER"])
+    if user:
+        patron_pid = user.pid
+    return patron_pid
+
+
+def create_order_line(record):
+    """Create an OrderLine."""
+    try:
+        item = get_item_by_barcode(record["barcode"])
+        document_pid = item.get("document_pid")
+        item_medium = item.get("medium")
+    except ItemMigrationError:
+        document_pid = MIGRATION_DOCUMENT_PID
+        item_medium = DEFAULT_ITEM_MEDIUM
+
+    new_order_line = dict(
+        document_pid=document_pid,
+        patron_pid=get_patron_pid(record),
+        recipient="LIBRARY",  # default or "PATRON"
+        medium=item_medium,
+        copies_ordered=1,  # default 1 because is required
+    )
+
+    total_price = get_cost(record)
+    if total_price:
+        new_order_line.update(total_price=total_price)
+
+    if record.get("budget_code"):
+        new_order_line.update(budget_code=record.get("budget_code"))
+
+    return new_order_line
+
+
+def migrate_order(record):
+    """Create a order record for ILS."""
+    new_order = dict(
+        legacy_id=record["legacy_id"],
+        order_lines=[create_order_line(record)],
+        status=get_status(record),
+    )
+
+    order_date = record["request_date"]
+    if order_date:
+        new_order.update(order_date=order_date.split("T")[0])
+
+    vendor_pid = get_vendor_pid_by_legacy_id(record["id_crcLIBRARY"])
+    new_order.update(vendor_pid=vendor_pid)
+
+    # Optional fields
+    grand_total = get_cost(record)
+    if grand_total:
+        new_order.update(grand_total=grand_total)
+
+    expected_delivery_date = record.get("expected_date")
+    if expected_delivery_date:
+        new_order.update(
+            expected_delivery_date=expected_delivery_date.split("T")[0]
+        )
+
+    received_date = record.get("arrival_date")
+    if received_date:
+        new_order.update(received_date=received_date.split("T")[0])
+
+    notes = get_acq_ill_notes(record)
+    if notes:
+        new_order.update(notes=notes)
+
+    return new_order
+
+
+def migrate_document_request(record):
+    """Create a document request record for ILS."""
+    state = "PENDING"
+    new_docreq = dict(
+        # legacy_id=record["legacy_id"],
+        patron_pid=get_patron_pid(record),
+        title="Migrated record, no title provided",
+        state=state,
+        request_type="LOAN",
+        medium="PAPER",
+    )
+
+    if record["status"] == "proposal-put aside":
+        state = "REJECTED"
+        reject_reason = record["library_notes"]
+        new_docreq.update(state=state, reject_reason=reject_reason)
+
+    note = get_acq_ill_notes(record)
+    if note:
+        new_docreq.update(note=note)
+
+    return new_docreq
+
+def is_order(record):
```

Comment:
We split on dump time, ill and acq records, but then we extract from acq the literature requests on load as you say. Let me note the json files have replicated information, since its all on table in cds.